### PR TITLE
Expand explanation of uses for dynamic blocks

### PR DESCRIPTION
**Pull request by jodamo5**
## Description
This update gives a clearer explanation of the uses for dynamic blocks. This update especially explains the need to use dynamic blocks if developers want their block updates to be immediately reflected on the front end of the site, instead of Gutenberg's validation process applying.

## How has this been tested?
No code has been changed - just the explanation about dynamic blocks.
For the second use-case example that I have added, I have tested this in Wordpress by creating a dynamic block, and have tested the values it saves into the database. 

## Types of changes
The existing description of dynamic blocks is extremely limited. The fuller explanation I have written gives far more breadth to the use of dynamic blocks.

### CODE REVIEW

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
 # Creating dynamic blocks
 
-Dynamic blocks are blocks that can change their content even if the post is not saved. One example from WordPress itself is the latest posts block. This block will update everywhere it is used when a new post is published.
+Dynamic blocks are blocks that build their structure and content on the fly when the block is rendered on the front end.
+
+There are two primary uses for dynamic blocks:
+1. Blocks where content should change even if a post has not been updated. One example from WordPress itself is the latest posts block. This block will update everywhere it is used when a new post is published.
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]:
```suggestion

1. Blocks where content should change even if a post has not been updated. One example from WordPress itself is the Latest Posts block. This block will update everywhere it is used when a new post is published.
```

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+2. Blocks where updates to the code (HTML, CSS, JS) should be immediately shown on the front end of the website. For example, if you update the HTML structure of a block by adding a new class, adding a div, or changing the layout in any other way, if you want these changes to be applied immediately on all occurrences of that block across the site, a dynamic block should be used. (If a dynamic block is not used then when block code is updated Guterberg's [validation process](https://developer.wordpress.org/block-editor/developers/block-api/block-edit-save/#validation) applies, causing users to see the validation message, "This block appears to have been modified externally").
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]: Would recommend not including the note around validation. This is a common occurrence but there are ways around it with block transforms (which should be better documented, admittedly).

```suggestion
2. Blocks where updates to the code (HTML, CSS, JS) should be immediately shown on the front end of the website. For example, if you update the structure of a block by adding a new class, adding an HTML element, or changing the layout in any other way, using a dynamic block ensures those changes are live across your entire site at the same time.
```

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+
+The general principle of dynamic blocks is that the built in `save` function should be returned as `null` so that Guternberg does not save the block HTML into the database. Instead Gutenberg will save only the [block attributes](https://developer.wordpress.org/block-editor/developers/block-api/block-attributes/) when the page is updated.  These attributes are then available to be used by the server-side rendering for displaying the block on the front end of your site.  (Alternatively, the return value of `save` could represent a cached copy of the block’s content to be shown only in case the plugin implementing the block is ever disabled.)
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]:
```suggestion
For many dynamic blocks, the `save` callback function should be returned as `null`, which tells the editor to save only the [block attributes](https://developer.wordpress.org/block-editor/developers/block-api/block-attributes/) to the database.  These attributes are then passed into the server-side rendering callback, so you can decide how to display the block on the front end of your site.

You can also save an HTML representation of the block. If you provide a server-side rendering callback, this HTML will be replaced with the output of your callback, but will be rendered if your block is deactivated or your render callback is removed.
```

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 4]
```diff
@@ -1,6 +1,15 @@
...
+
+Block attributes can be used for any content or setting you want to save for that block. In the first example above, with the latest posts block, the number of latest posts you want to show could be saved as an attribute. Or in the second example, attributes can be used for each piece of content you want to show in the front end - such as heading text, paragraph text, an image, a URL, etc.
+
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]:
```suggestion
```

**Review by chrisvanpatten — COMMENTED** [review]
I've left some suggestions, large and small, which attempt to break up the content and also clarify a few pieces of how server-rendered blocks work. Please let me know if you have any questions!

**Comment by jodamo5** [comment]
Hi. As a first-time contributor I'm not sure if there is anything else I'm supposed to do here. I have accepted all the comments from @chrisvanpatten and have recommitted it 6 days ago. But nothing has happened since.  It says another review is required, and there are 5 pending reviewers. Can someone please review and merge these changes? Or is there something else I'm supposed to do?  Thanks.

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 5]
```diff
@@ -1,6 +1,16 @@
 # Creating dynamic blocks
 
-Dynamic blocks are blocks that can change their content even if the post is not saved. One example from WordPress itself is the latest posts block. This block will update everywhere it is used when a new post is published.
+Dynamic blocks are blocks that build their structure and content on the fly when the block is rendered on the front end.
+
+There are two primary uses for dynamic blocks:
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]:
```suggestion
There are two primary uses for dynamic blocks:

```

**Comment by chrisvanpatten** [comment]
@jodamo5 Sorry, one more quick fix then I'm happy to approve and merge.

**Comment by jodamo5** [comment]
@chrisvanpatten One of the author tests failed in the build apparently, which seems strange. Can you provide guidance for next steps please?

**Comment by youknowriad** [comment]
restarted the build, it should be unrelated.

**Comment by jodamo5** [comment]
Great.  All the checks passed.  Can someone merge this please?

**Review by chrisvanpatten — APPROVED** [review]

**Comment by chrisvanpatten** [comment]
Thanks for your patience, @jodamo5. Looks great; excited to have this improvement in place.

**Pull request merged**